### PR TITLE
Admin Route Extensibility

### PR DESCRIPTION
**Pull request by HexaField**
## Summary

Projects can take advantage of the webappInjection API in xrengine.config.ts to add new routes

### <samp>🤖 Generated by Copilot at cfbf771</samp>

This pull request improves the admin dashboard functionality by using a new state for the allowed admin routes, refactoring the admin routes logic, and removing unused or outdated code. It also enhances the performance and readability of the admin dashboard components by using code splitting, simplifying the rendering logic, and using a new Redirect component. The main files affected are `.env.local.default`, `appconfig.ts`, `adminRoutes.tsx`, `allowedRoutes.tsx`, `AllowedAdminRoutesState.tsx`, `DefaultAdminRoutes.tsx`, `DashboardMenuItem/index.tsx`, and `Redirect.tsx`.

## References
closes #_insert number here_

### <samp>🤖 Generated by Copilot at cfbf771</samp>

*  Remove unused or deprecated config variables from `.env.local.default` file ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-fc461757101c0a3bfa1c7c5b1bd17dbc296785124584e6d61bfee5a879aa89a3L200-L206), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-fc461757101c0a3bfa1c7c5b1bd17dbc296785124584e6d61bfee5a879aa89a3L215-L224))
*  Rename `DashboardItems` file to `DefaultAdminRoutes` and export an object of default admin routes with name, scope, component, access, and icon properties ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-4dff24ac8149a278e956f2c48558c8eaaa98d7ceb3e447eb5fbef16e07bdcdceL26-R152))
*  Add a new file `AllowedAdminRoutesState` that defines a state for the allowed admin routes using Hyperflux library ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-7c29e05f4ec5f6fbb8b6e2af6aae61ca5febb1dc613f585d239f66f6602ada31R1-R41))
*  Modify `AdminRoutes` file to import and use the `AllowedAdminRoutesState` and the `DefaultAdminRoutes` to determine the user's access to each admin route based on their scopes ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-555e2fa4a2c45d91dff6f12d62ddab3276f060f05c9622f2c7f7d63512d56d0eL38-R40), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-555e2fa4a2c45d91dff6f12d62ddab3276f060f05c9622f2c7f7d63512d56d0eL50-R72), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-555e2fa4a2c45d91dff6f12d62ddab3276f060f05c9622f2c7f7d63512d56d0eL94-R82))
*  Modify `allowedRoutes` file to use the allowed routes state from the `AllowedAdminRoutesState`, instead of the props passed from the `adminRoutes` file, and to render the matched route using the pathname and the access property of each route ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-e6267de35503fc74a4695520f18393bc7a757d63ac0356dc5ec8bf41f9c0a3f6L27-R53))
*  Add a new file `Redirect` that defines a Redirect component using the RouterState from the RouterService, and use it to redirect the user to a different route if the current route has a redirect property in the `AllowedAdminRoutesState` ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-0b5a4682138f3a0c94faf61ec0eb63152832d07bc24beac64adfcbc4ee386f06R1-R34))
*  Modify `DashboardMenuItem` file to use the `AllowedAdminRoutesState` to render the sidebar items, instead of the useState and useEffect hooks, and to filter and map the sidebar items using the Object.entries and the access property of each route ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-ad2422d7e19c8dc98010720699277098d82345740225d057b8830a3713d56ca5L26-R31), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-ad2422d7e19c8dc98010720699277098d82345740225d057b8830a3713d56ca5L45-R60))
*  Remove the scopes object and property from the `appconfig` file, as they are now defined in the database and the user service ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-3c36f28a0af9a4f137dde08e430d1279036aa0b3d214200d98b069c61bfa2100L378-L385), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-3c36f28a0af9a4f137dde08e430d1279036aa0b3d214200d98b069c61bfa2100L412))
*  Remove the Analytics component import from the `AdminRoutes` and `DashboardMenuItem` files, as it is no longer a separate component ([link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-555e2fa4a2c45d91dff6f12d62ddab3276f060f05c9622f2c7f7d63512d56d0eL38-R40), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-555e2fa4a2c45d91dff6f12d62ddab3276f060f05c9622f2c7f7d63512d56d0eL50-R72), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-555e2fa4a2c45d91dff6f12d62ddab3276f060f05c9622f2c7f7d63512d56d0eL94-R82), [link](https://github.com/EtherealEngine/etherealengine/pull/9043/files?diff=unified&w=0#diff-ad2422d7e19c8dc98010720699277098d82345740225d057b8830a3713d56ca5L26-R31))

### <samp>🤖 Generated by Copilot at cfbf771</samp>

> _To make the admin routes more dynamic_
> _We used Hyperflux and React.lazy_
> _We added `AllowedAdminRoutesState`_
> _And a `Redirect` component that's great_
> _And cleaned up the config and `appconfig.ts`_

## QA Steps
_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

## Checklist
- If this PR is still a WIP, convert to a draft
- When this PR is ready, mark it as "Ready for review"
- [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- Changes have been manually QA'd 
- Changes reviewed by at least 2 approved reviewers

### CODE REVIEW

File: packages/client-core/src/admin/AllowedAdminRoutesState.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import { defineState } from '@etherealengine/hyperflux'
+import React from 'react'
+
+export type AdminRouteStateType = {
+  name: string
+  scope: string
+  redirect?: string
+  component: React.LazyExoticComponent<() => JSX.Element>
+  access: boolean
```
Comment (hanzlamateen) [inline packages/client-core/src/admin/AllowedAdminRoutesState.tsx:34]: What this access thing is for?

File: packages/client-core/src/admin/DefaultAdminRoutes.tsx [hunk 2]
```diff
@@ -23,84 +23,130 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
...
-  allowedRoutes.benchmarking && {
-    name: 'user:dashboard.benchmarking',
-    path: '/admin/benchmarking',
-    icon: <Icon type="Timeline" style={{ color: 'white' }} />
-  },
-  allowedRoutes.settings && {
+  settings: {
     name: 'user:dashboard.setting',
-    path: '/admin/settings',
+    scope: 'settings',
```
Comment (hanzlamateen) [inline packages/client-core/src/admin/DefaultAdminRoutes.tsx:140]: Is there a way we can control subsettings in admin panel? Like a user can only manage client-settings and not the other settings.

File: packages/client-core/src/admin/AllowedAdminRoutesState.tsx (hunk 1 above)
Comment (HexaField) [reply packages/client-core/src/admin/AllowedAdminRoutesState.tsx:34]: this is the state that stores whether or not a user has access to the given route

File: packages/client-core/src/admin/DefaultAdminRoutes.tsx (hunk 2 above)
Comment (HexaField) [reply packages/client-core/src/admin/DefaultAdminRoutes.tsx:140]: hmm, we will need separate scopes for each setting type in that case anyway, so we should drive it from that

**Review by hanzlamateen — APPROVED** [review]

**Pull request merged**